commonv5/ts/sonarqube/Analysis.ts

```diff
@@ -44,145 +38,124 @@ const separatorStyle = `
+      this.getHtmlQualityGateSection(),
+      this.getHtmlSeparator(),
+      this.getHtmlQualityGateDetailSection(),
+      this.getHtmlDashboardLink(),
+      this.getHtmlWarnings(),
     ]
       .join(" \r\n")
       .trim();
   }
 
-  private getSeparator() {
-    return `<div style="${separatorStyle}"></div>`;
+  private getHtmlSeparator() {
+    return `
+    <div style="${separatorStyle}"></div>
+    `;
   }
 
-  private getQualityGateSection() {
-    return `<div style="${textStyle()}">
-      <span>${this.projectName ? this.projectName + " " : ""}Quality Gate</span>
-      <span style="${measureStyle(this.getQualityGateColor())}">
-        ${formatMeasure(this.status, "LEVEL")}
+  private getHtmlQualityGateSection() {
+    return `
+    <div style="${textStyle()}">
+      <span>${this.result.projectName ? this.result.projectName + " " : ""}Quality Gate</span>
+      <span style="${measureStyle(qualityGateColor(this.projectStatus.status))}">
+        ${formatMeasure(this.projectStatus.status, "LEVEL")}
       </span>
-    </div>`;
+    </div>
+    `;
   }
 
-  private getQualityGateDetailSection() {
+  private getHtmlQualityGateDetailSection() {
     const failedConditions = this.getFailedConditions();
-    if (!this.metrics || !["WARN", "ERROR"].includes(this.status) || failedConditions.length <= 0) {
+    if (
+      !this.result.metrics ||
+      !["WARN", "ERROR"].includes(this.projectStatus.status) ||
+      failedConditions.length <= 0
+    ) {
       return "";
     }
 
     const rows = failedConditions.map((condition) => {
-      const metric = this.metrics && this.metrics.getMetricByKey(condition.metricKey);
+      const metric = this.result.metrics?.getMetricByKey(condition.metricKey);
       if (!metric) {
         return null;
       }
 
       const threshold =
         condition.status === "WARN" ? condition.warningThreshold : condition.errorThreshold;
-      return `<li style="list-style: none; margin-top: 13px;">
+      return `
+      <li style="list-style: none; margin-top: 13px;">
         <span>${metric.name}</span>
-        <span style="${measureStyle(this.getQualityGateColor())}">
+        <span style="${measureStyle(qualityGateColor(this.projectStatus.status))}">
           ${formatMeasure(condition.actualValue, metric.type)}</span>
         <span>(required ${
           metric.type !== "RATING" ? formatMeasure(condition.comparator, "COMPARATOR") + " " : ""
         }${formatMeasure(threshold, metric.type)})</span>
-      </li>`;
+      </li>
+      `;
     });
 
-    return `<ul style="${textStyle()}">
-        ${rows.join(" \r\n").trim()}
-    </ul>\r\n\r\n`; // 2 carriage returns to prevent any malformed summary results
+    return `
+    <ul style="${textStyle()}">
+      ${rows.join(" \r\n").trim()}
+    </ul>\r\n\r\n
+    `; // 2 carriage returns to prevent any malformed summary results
   }
 
-  private getDashboardLink() {
-    if (!this.dashboardUrl) {
+  private getHtmlDashboardLink() {
```

Comment:
Prefixed all methods returning html with `getHtml` so it's obvious what it's doing